src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = {};
+  const fromSeparator = fromFormat[fromFormat.length - 1];
```

Comment:
The logic assumes that the last element of `fromFormat` is the separator. Ensure that this assumption is valid for all use cases. If the separator is not always the last element, this logic might fail.